Adjust the style of the first column of the attribute view

Zuoqiu-Yingyi opened the pull request:
* [x] Please commit to the dev branch

REL: #9671

- 调整属性视图首列/钉住的列样式
  Adjust the style of the first/pinned column of the attribute view
- 为了区分 `.av__firstcol` 中的复选框图标与其他图标, 添加 `.av__check` CSS 类
  To distinguish the checkbox icon in the `.av__firstcol` from other icons, add the `.av__check` CSS class

Zuoqiu-Yingyi committed:
:art: Adjust the style of the column pinned to the attribute view

Vanessa219 commented:
麻烦截图标注一下`调整属性视图首列/钉住的列样式`前后的区别。

Zuoqiu-Yingyi commented:
> 麻烦截图标注一下`调整属性视图首列/钉住的列样式`前后的区别。

主要是调整了一下 DOM 结构, 由

```html
<div class="av__colsticky av__firstcol">
    <svg><use xlink:href="#iconUncheck"></use></svg>
</div>
```

调整为了

```html
<div class="av__colsticky">
    <div class="av__firstcol">
        <svg class="av__check"><use xlink:href="#iconUncheck"></use></svg>
    </div>
</div>
```

外观基本没啥区别, 除了最后两行在 `av__firstcol` 内添加了图标 (为了与其他行的首列宽度保持一致)
![image](https://github.com/siyuan-note/siyuan/assets/49649786/5eca0786-2c09-49c0-892c-d70c581a460e)

Zuoqiu-Yingyi committed:
Merge branch 'dev' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi committed:
:art: Adjust the style of first column icon

Zuoqiu-Yingyi committed:
Update index.ts

Zuoqiu-Yingyi commented:
![image](https://github.com/siyuan-note/siyuan/assets/49649786/b32804f6-cc81-4265-b17d-dd688d706734)

Vanessa219 commented:
这涉及到交互和 UI 的改动，破坏了原有的设计，没有办法进行合并。

Zuoqiu-Yingyi commented:
> 这涉及到交互和 UI 的改动，破坏了原有的设计，没有办法进行合并。

哪些改动涉及了交互和 UI 的改动?

Vanessa219 commented:
图中框内部分
![image](https://github.com/siyuan-note/siyuan/assets/970828/393e6bdc-e626-48dd-9250-eabde081f825)

Zuoqiu-Yingyi commented:
> 图中框内部分
> ![image](https://github.com/siyuan-note/siyuan/assets/970828/393e6bdc-e626-48dd-9250-eabde081f825)
> 

红色框改进为了更加方便用户在需要点击首列控件时定位点击位置

蓝色框为了避免用户误触控件

绿色框的更改具体是指？

Zuoqiu-Yingyi committed:
Merge branch 'dev' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi commented:
已兼容 `v2.11.2-dev5`

Vanessa219 commented:
蓝色指的是该位置的距离
<img width="296" alt="image" src="https://github.com/siyuan-note/siyuan/assets/970828/e33d4712-3efa-4284-802a-3e01ddee0f8c">

目前交互主要参照 notion

Zuoqiu-Yingyi committed:
:art: Adjust the style of add row in attribute view

https://github.com/siyuan-note/siyuan/pull/9830?notification_referrer_id=NT_kwDOAvWYerM4NjY1MTYyMTg0OjQ5NjQ5Nzg2#issuecomment-1853268143

Zuoqiu-Yingyi commented:
> 蓝色指的是该位置的距离 <img alt="image" width="296" src="https://private-user-images.githubusercontent.com/970828/290077936-e33d4712-3efa-4284-802a-3e01ddee0f8c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0NTc5NzEsIm5iZiI6MTcwMjQ1NzY3MSwicGF0aCI6Ii85NzA4MjgvMjkwMDc3OTM2LWUzM2Q0NzEyLTNlZmEtNDI4NC04MDJhLTNlMDFkZGVlMGY4Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxM1QwODU0MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YjUzNjZmNjBmYWI2YjAxNTBjMzQ2NjgwMjhhMzI4OTQwZDA1MDJiODU4MTM3YjVkMzczMmE2NWIwNzc3NTg1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.vJ1FAmYk400VR7zxRK4MrsOUg4R9oHgh0wi5hA1fYGc">
> 
> 目前交互主要参照 notion

已调整
![image](https://github.com/siyuan-note/siyuan/assets/49649786/dd7611ab-5822-43f1-90b7-c09c0effdd56)

Vanessa219 commented:
> 目前交互主要参照 notion 

未经过讨论决定的改动不会进行合并。

Zuoqiu-Yingyi commented:
> 未经过讨论决定的改动不会进行合并。

那么下一个议题是关于哪一处改动的讨论？

Zuoqiu-Yingyi committed:
Merge branch 'master' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi committed:
Merge branch 'dev' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi committed:
Merge branch 'master' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi commented:
什么时候继续展开讨论👀?

zxhd863943427 commented:
不太懂这个改进涉及什么方面

Zuoqiu-Yingyi commented:
> 不太懂这个改进涉及什么方面

就是 DOM 由 `<div class="av__colsticky av__firstcol"></div>` 结构转换为 `<div class="av__colsticky"><div class="av__firstcol"></div></div>` 以统一样式

Zuoqiu-Yingyi committed:
Merge branch 'dev' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi committed:
Update protyle.ts

Zuoqiu-Yingyi committed:
Merge branch 'master' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi committed:
Merge branch 'master' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi committed:
Merge tag 'v2.12.1-dev3' into perf/attr-view-pin-col-sticky

Zuoqiu-Yingyi committed:
Merge branch 'master' into perf/attr-view-pin-col-sticky

Vanessa219 commented:
看了下修改后的结构应该是比较合理的，但由于修改了部分交互，基于以下原因建议还原已有交互后重新进行提交。非常感谢。

1.  鼠标浮动在行上时显示首例的勾选框会分散用户注意力，造成更多不必要的干扰。

https://github.com/siyuan-note/siyuan/assets/970828/d37b3f01-0c4c-4344-b765-084cee0291d6

2. 没有进行列计算时，最后一行的竖线比较突兀
<img width="1449" alt="image" src="https://github.com/siyuan-note/siyuan/assets/970828/85abeb5f-82be-453f-885b-1fbe40ac14e5">

3. 为了简洁，原有设计在第一列后没有竖线。但社区有用户提过像 wolai 一样提供所有的线条，为此建议使用主题或代码片段进行实现。
<img width="1511" alt="image" src="https://github.com/siyuan-note/siyuan/assets/970828/6234b909-21a9-4122-a45c-dc36c5d2d125">

4. 两个加号在一起比较奇怪
<img width="1397" alt="image" src="https://github.com/siyuan-note/siyuan/assets/970828/10a68e31-9768-451b-a53d-64f26ba388d3">

5. 按钮高亮样式过于粗壮
<img width="1443" alt="image" src="https://github.com/siyuan-note/siyuan/assets/970828/10f084da-a825-4aa4-be06-85164ecf32f4">

6. 这两个按钮不可以点击，但是鼠标移动上去会高亮，并且为点击手势
<img width="1477" alt="image" src="https://github.com/siyuan-note/siyuan/assets/970828/c5c4cec6-e8b6-4a7e-9388-04b38dbcbfa4">

Zuoqiu-Yingyi commented:
@Vanessa219 这些地方可能还需要进一步讨论

> 1. 鼠标浮动在行上时显示首例的勾选框会分散用户注意力，造成更多不必要的干扰。

鼠标浮动在行上时显示首列的勾选框理由请参考 https://github.com/siyuan-note/siyuan/pull/9772#issuecomment-1833222888, 这里不再赘述
类似于 Windows 文件资源管理器, 这实际上 **不会** 分散用户注意力
![image](https://github.com/siyuan-note/siyuan/assets/49649786/6cf0be51-f114-4500-868e-7178557415d2)

---
> 2. 没有进行列计算时，最后一行的竖线比较突兀
> 3. 为了简洁，原有设计在第一列后没有竖线。但社区有用户提过像 wolai 一样提供所有的线条，为此建议使用主题或代码片段进行实现。

首列没有竖线但其依然存在, 且在横向滚动时会遮挡其他内容, 会被用户视为 bug (反直觉)

---
> 4. 两个加号在一起比较奇怪
> 6. 这两个按钮不可以点击，但是鼠标移动上去会高亮，并且为点击手势

这里需要删除这两个图标还是优化这两个图标 (更换图标+调整鼠标手势)?

Vanessa219 commented:
以上几点没有争议的可以先进行修改，其他可以再看看大部分用户的需求。

Zuoqiu-Yingyi commented:
> 以上几点没有争议的可以先进行修改，其他可以再看看大部分用户的需求。

@Vanessa219 这里有未决的问题

> 4. 两个加号在一起比较奇怪
> 5. 这两个按钮不可以点击，但是鼠标移动上去会高亮，并且为点击手势

这里需要删除这两个图标还是优化这两个图标 (更换图标+调整鼠标手势)?
